### PR TITLE
multi-city graph creation

### PR DESCRIPTION
**Pull request by lkenigsb**
Screen Shot 2023-04-27 at 3 45 52 PM
Screen Shot 2023-04-27 at 3 46 05 PM

### CODE REVIEW

File: src/main/java/kenigsberg/weather/WeatherFrame.java [hunk 1]
```diff
@@ -1,18 +1,73 @@
...
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl("https://api.openweathermap.org/")
+                .addConverterFactory(GsonConverterFactory.create())
+                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
+                .build();
+
+
+        WeatherService service = retrofit.create(WeatherService.class);
+
+        FiveDayWeather fiveDayWeather = service.getFiveDayWeather("Staten Island").blockingFirst();
```
Comment (andrewoid) [inline src/main/java/kenigsberg/weather/WeatherFrame.java]: blockingFirst() is only suitable for unit tests.

Do you remember how to do this with subscribe ()?

File: src/main/java/kenigsberg/weather/WeatherFrame.java [hunk 2]
```diff
@@ -1,18 +1,73 @@
...
+
+        CurrentWeatherView currentWeatherView = new CurrentWeatherView(fiveDayWeather);
+        currentWeatherView.setForecast(fiveDayWeather);
```
Comment (andrewoid) [inline src/main/java/kenigsberg/weather/WeatherFrame.java]: You shouldn't pass anything into the constructor.

File: src/main/java/kenigsberg/weather/WeatherFrame.java [hunk 3]
```diff
@@ -1,18 +1,73 @@
...
+        });
+
+
+        setContentPane(mainPanel);
+
+    }
+
+    /*
+
+    EARTHQUAKE GUI
```
Comment (andrewoid) [inline src/main/java/kenigsberg/weather/WeatherFrame.java]: This was never an earthquake gui

**Review by andrewoid — COMMENTED** [review]
Put a picture of your gui in the PR

File: src/main/java/kenigsberg/weather/WeatherFrame.java (hunk 1 above)
Comment (lkenigsb) [reply src/main/java/kenigsberg/weather/WeatherFrame.java:26]: I wrote it like this, but it's not working... Did I write the .subscribe() wrong?

```
Disposable disposable = service.getFiveDayWeather("Staten Island")
                .subscribeOn(Schedulers.io())
                .observeOn(Schedulers.newThread())
                .subscribe(
                        fiveDayWeather -> {
                            currentWeatherView = new CurrentWeatherView(fiveDayWeather);
                            currentWeatherView.setForecast(fiveDayWeather);
                            mainPanel.add(currentWeatherView, BorderLayout.CENTER);

                        },
                        Throwable::printStackTrace

                );
```

File: src/main/java/kenigsberg/weather/WeatherFrame.java [hunk 4]
```diff
@@ -8,11 +8,77 @@
...
+        JTextField location = new JTextField("Staten Island");
+        topPanel.add(location);
+
+        JButton button = new JButton("Submit");
+        topPanel.add(button);
+
+        mainPanel.add(topPanel, BorderLayout.NORTH);
+
+
+        mainPanel.add(currentWeatherView, BorderLayout.CENTER);
```
Comment (andrewoid) [inline src/main/java/kenigsberg/weather/WeatherFrame.java:40]: At this point, `currentWeatherView` is null.

File: src/main/java/kenigsberg/weather/RunInfo.java [hunk 5]
```diff
@@ -1,11 +1,19 @@
...
+
 import java.io.IOException;
 
 public class RunInfo {
     public static void main(String[] args) throws IOException {
 
-        WeatherFrame earthquakeFrame = new WeatherFrame();
-        earthquakeFrame.setVisible(true);
+        //WeatherFrame earthquakeFrame = new WeatherFrame();
+        //earthquakeFrame.setVisible(true);
```
Comment (andrewoid) [inline src/main/java/kenigsberg/weather/RunInfo.java]: Remove these commented out lines.

File: src/main/java/kenigsberg/weather/WeatherFrame.java [hunk 6]
```diff
@@ -8,11 +8,79 @@
...
+        Disposable disposable = service.getFiveDayWeather("Staten Island")
+                .subscribeOn(Schedulers.io())
+                .observeOn(Schedulers.newThread())
+                .subscribe(
+                        fiveDayWeather -> {
+                            currentWeatherView.setForecast(fiveDayWeather);
+                        },
+                        Throwable::printStackTrace
+
+                );
```
Comment (andrewoid) [inline src/main/java/kenigsberg/weather/WeatherFrame.java]: Can you remove the code duplication between here and line 65?

Comment (lkenigsb) [reply src/main/java/kenigsberg/weather/WeatherFrame.java]: How can I do that? Is there a way to call the already-created subscribe?

Comment (andrewoid) [reply src/main/java/kenigsberg/weather/WeatherFrame.java]: No. You won't call the already created subscribe.

You want the same code to be run in two places.
How can you do that?

Comment (lkenigsb) [reply src/main/java/kenigsberg/weather/WeatherFrame.java]: Oh! To create a method with that code? And call it where needed?

Comment (andrewoid) [reply src/main/java/kenigsberg/weather/WeatherFrame.java]: yes

**Review by andrewoid — APPROVED** [review]

**Pull request merged**